packages/e2e/test/wallet/SingleAddressWallet/staking-rewards.test.ts

```diff
@@ -0,0 +1,200 @@
+    faucetProvider = await faucetProviderFactory.create(env.FAUCET_PROVIDER, env.FAUCET_PROVIDER_PARAMS, logger);
+
+    await faucetProvider.start();
+
+    const healthCheck = await faucetProvider.healthCheck();
+
+    if (!healthCheck.ok) throw new Error('Faucet provider could not be started.');
+  });
+
+  afterAll(async () => {
+    await faucetProvider.close();
+  });
+
+  beforeEach(async () => {
+    wallet1 = await getWallet({ env, logger, name: 'Sending Wallet', polling: { interval: 50 } });
+    wallet2 = await getWallet({ env, logger, name: 'Receiving Wallet', polling: { interval: 50 } });
+  });
+
+  afterEach(() => {
+    wallet1.wallet.shutdown();
+    wallet2.wallet.shutdown();
+  });
+
+  it('will do tADA transfer between two wallets.', async () => {
+    // Arrange
+    const amountFromFaucet = 100_000_000;
+    const tAdaToSend = 50_000_000n;
+
+    const saWallet1 = wallet1.wallet;
+    const saWallet2 = wallet2.wallet;
+
+    await firstValueFrom(saWallet1.syncStatus.isSettled$.pipe(filter((isSettled) => isSettled)));
+    await firstValueFrom(saWallet2.syncStatus.isSettled$.pipe(filter((isSettled) => isSettled)));
+
+    const [{ address: sendingAddress }] = await firstValueFrom(saWallet1.addresses$);
+    const [{ address: receivingAddress }] = await firstValueFrom(saWallet2.addresses$);
+
+    // Act
+
+    logger.info(`Address ${sendingAddress.toString()} will be funded with ${amountFromFaucet} tLovelace.`);
+
+    // Request 100 tADA from faucet. This will block until the transaction is in the ledger,
+    // and has the given amount of confirmation, which means the funds can be used immediately after
+    // this call.
+    await faucetProvider.request(sendingAddress.toString(), amountFromFaucet, 1);
+
+    // Wait until wallet one is aware of the funds.
+    await firstValueFrom(saWallet1.balance.utxo.total$.pipe(filter(({ coins }) => coins >= amountFromFaucet)));
+
+    logger.info(
+      `Address ${sendingAddress.toString()} will send ${tAdaToSend} lovelace to address ${receivingAddress.toString()}.`
+    );
+
+    // Send 50 tADA to second wallet.
+    const txBuilder = buildTx({ logger, observableWallet: saWallet1 });
+    const txOut = txBuilder.buildOutput().address(receivingAddress).coin(tAdaToSend).toTxOut();
+    const unsignedTx = await txBuilder.addOutput(txOut).build();
+    assertTxIsValid(unsignedTx);
+    const signedTx = await unsignedTx.sign();
+    await signedTx.submit();
+
+    // Wait until wallet two is aware of the funds.
+    await firstValueFrom(saWallet2.balance.utxo.total$.pipe(filter(({ coins }) => coins >= tAdaToSend)));
+
+    // Search chain history to see if the transaction is there.
+    const txFoundInHistory = await firstValueFrom(
+      saWallet2.transactions.history$
+        .pipe(filter((txs) => txs.filter((tx) => tx.id === signedTx.tx.id).length === 1))
+        .pipe(map((txs) => (txs.length > 0 ? txs.find((tx) => tx.id === signedTx.tx.id) : undefined)))
```

Comment:
Minor improvement: no need for 2 `.pipe`, also no need to iterate it twice. This will also result in a variable that can't be undefined, so will eliminate the need for `txFoundInHistory?`
```suggestion
        .pipe(
          map((txs) => txs.find((tx) => tx.id === signedTx.tx.id)),
          filter(isNotNil),
          take(1)
        )
```